packages/react-client/src/__tests__/ReactFlight-test.js

```diff
@@ -164,6 +185,49 @@ describe('ReactFlight', () => {
     });
   });
 
+  it('should trigger the inner most error boundary inside a client component', () => {
+    function ServerComponent() {
+      throw new Error('This was thrown in the server component.');
+    }
+
+    function ClientComponent({children}) {
+      // This should catch the error thrown by the server component, even though it has already happened.
+      // We currently need to wrap it in a div because as it's set up right now, a lazy reference will
+      // throw during reconciliation which will trigger the parent of the error boundary.
+      // This is similar to how these will suspend the parent if it's a direct child of a Suspense boundary.
+      // That's a bug.
```

Comment:
@acdlite This is related to the problem were throwing during reconciliation triggers the nearest boundary. These are lazy *elements*. Not lazy types. So they throw here and not during beginWork. Since we don't have the key and stuff necessarily to complete the reconciliation. https://github.com/facebook/react/blob/master/packages/react-reconciler/src/ReactChildFiber.new.js#L1251